bpf/include/vmlinux.h

```diff
@@ -1411,6 +1411,8 @@ struct range {
 	u64 end;
 };
 
+#if defined(__TARGET_ARCH_x86)
```

Comment:
I think the kernel doesn't have these right?

Probably the best way to do it is to have different vmlinux.h per architecture, then we compile the right one according to the TARGET_ARCH, something like bpf/include/{amd64|arm64}/vmlinux.h ?

Reply:
I'm not sure to understand your question, but this is not a new thing from this PR, we are using this in libbpf currently for reading the registers https://github.com/cilium/tetragon/blob/main/bpf/libbpf/bpf_tracing.h#L18 of calls.

I mostly reused this because it's just more simple.

Reply:
The vmlinux header is dumped from the current running kernel using something like:
`bpftool btf dump file /sys/kernel/btf/vmlinux format c > vmlinux.h` 

Those registers are not the only things that may differ...  IIRC as an example on ARM char is unsigned by default, and on x86 it is not defined (depends on implementation)... etc

libbpf uses later CORE to locate fields etc but that's another story. So we should not mix it here, for each arch we must generate a proper vmlinux file, let's ping @jrfastab on to know which kernel config or kernel stock image they used to generate the current one.

Reply:
Yes yes I see your point but again what's the benefit of using different files and not using the preprocessor macros directly? Tidiness?

Reply:
btw how do you know which structs are accessed using CORE and not? (this one is kinda obvious since it's completely arch-specific but is there a way?). Just to be clear it compiles like that at the moment, so it's a matter of putting things in the right place?

Should I put this part in a separate header file?
```c
#if defined(__TARGET_ARCH_x86)

struct pt_regs {
	long unsigned int r15;
	long unsigned int r14;
	long unsigned int r13;
	long unsigned int r12;
	long unsigned int bp;
	long unsigned int bx;
	long unsigned int r11;
	long unsigned int r10;
	long unsigned int r9;
	long unsigned int r8;
	long unsigned int ax;
	long unsigned int cx;
	long unsigned int dx;
	long unsigned int si;
	long unsigned int di;
	long unsigned int orig_ax;
	long unsigned int ip;
	long unsigned int cs;
	long unsigned int flags;
	long unsigned int sp;
	long unsigned int ss;
};

#elif defined(__TARGET_ARCH_arm64)

struct user_pt_regs {
	__u64 regs[31];
	__u64 sp;
	__u64 pc;
	__u64 pstate;
};

struct pt_regs {
	union {
		struct user_pt_regs user_regs;
		struct {
			u64 regs[31];
			u64 sp;
			u64 pc;
			u64 pstate;
		};
	};
	u64 orig_x0;
	s32 syscallno;
	u32 unused2;
	u64 sdei_ttbr1;
	u64 pmr_save;
	u64 stackframe[2];
	u64 lockdep_hardirqs;
	u64 exit_rcu;
};

#endif
```

Reply:
CORE accesses in our BPF code use the semantic `_(&xyz)` so you can check the BPF code directly.

I think doing a ifdef like this in a specific header is fine. I don't have anything better off-hand.